www/js/diary/current.js

```diff
@@ -110,115 +173,95 @@
-                                    {text: "Cancel",
-                                     action: cancelTempEntry}];
-    Logger.log("About to call ionicActionSheet.show");
-    $ionicActionSheet.show({titleText: "lat: "+latlng.lat.toFixed(6)+
-              ", lng: " + latlng.lng.toFixed(6),
-          cancel: function() {
-            cancelTempEntry(m, ts, map);
-          },
-          buttons: safe_suck_cancel_actions,
-          buttonClicked: function(index, button) {
-            button.action(m, ts, map);
-            Logger.log("Clicked button "+button.text+" at index "+index);
-            return true;
-          }
+  var getServerIncidents = function() {
+      console.log(incidentServerCalldata);
+      $http.post("https://e-mission.eecs.berkeley.edu/result/heatmap/incidents/local_date", incidentServerCalldata).then(function(res){
+      console.log(res);
+      if(res.data.incidents.length > 0) {
+        addIncidents(res.data.incidents, _map);
+      }
+    }, function(error){
+      console.log("Error when getting incidents");
+      console.log(error);
     });
-      
   };
-  
-  $scope.parse = function() {
-    console.log("About to get map"); 
-    var _map;
-    leafletData.getMap('current').then(function(map) {
-      console.log("success, got map "+ map);
-      _map = map;
-      return db.getAllSensorData(BACKGROUND_LOCATION);
-    }).then(function(result) {
+
+  var marker;
+  $scope.showIncidentSheet = function() {
+    db.getAllSensorData(BACKGROUND_LOCATION).then(function(result) {
             both = [result[0].data.latitude, result[0].data.longitude];
             var ts = result[0].data.ts;
-            addmarker(_map, ts, both);
+            var latlng = L.latLng(both);
+            $scope.features = [];
+            marker = L.circleMarker(latlng);
+            console.log(marker);
+            PostTripManualMarker.showSheet($scope.features, latlng, ts, marker, _map);
     })
     .catch(function(error) {
       console.log("error while getting map current from leafletData");
     });
   };
 
+  $scope.$watch('features', function(newVal, oldVal){
+    if(newVal.length === 1) {
+      addIncidentLayer(newVal[0].properties.stress, marker, _map);
+    }
+  }, true);
+
   var mapRunning;
-  var realTimeReload = function() {
+  var gettingIncidents;
+  var refreshTripLoop = function() {
     refreshTrip();
-    mapRunning = setTimeout(realTimeReload, 1000);
+    mapRunning = setTimeout(refreshTripLoop, 1000); //refresh every second
   };
 
-  $scope.$on('$ionicView.enter', function() {
-    realTimeReload();
-  });
+  var getIncidentsLoop = function() {
+    getServerIncidents();
+    gettingIncidents = setTimeout(getIncidentsLoop, 1000*60); //refresh every minute
+  };
 
-  $scope.$on('$ionicView.leave', function() {
-    clearTimeout(mapRunning);
-  });
+  leafletData.getMap('current').then(function(map) {
+    _map = map;
+    _map.removeControl(_map.zoomControl);
+    $scope.$on('$ionicView.enter', function() {
```

Comment:
why are these inside `getMap`?
they don't need the map, and isn't it bad if you register for the enter event a little late? Maybe it will be fired before you register for it...